From 0d1ac45b738195db5e2785531db713328f547e60 Mon Sep 17 00:00:00 2001
From: FaramosCZ <mschorm@redhat.com>
Date: Tue, 29 Jun 2021 18:08:36 +0200
Subject: [PATCH] CMake: Enhance logic around GLM (#133)

* CMake: Add a status message regarding GLM used (system / bundled)

* CMake: Introduce logic and variable WITH_GLM to control which GLM version should be used (system / bundled / auto)
---
 CMakeLists.txt | 18 ++++++++++++++++--
 1 file changed, 16 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7b96f83..200de46 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -89,8 +89,22 @@ if(NOT ${SFML_FOUND})
 endif()
 
 
-find_package(glm QUIET)
-if(NOT ${glm_FOUND})
+# GLM - OpenGL Mathematic library
+SET(WITH_GLM "auto" CACHE STRING "Which GLM to use (possible values are 'bundled', 'system' or 'auto')")
+
+if(WITH_GLM STREQUAL "system" OR WITH_GLM STREQUAL "auto")
+    find_package(glm QUIET)
+
+    if(glm_FOUND)
+        message(STATUS "GLM version used: SYSTEM")
+    elseif(WITH_GLM STREQUAL "system")
+        message(FATAL_ERROR "GLM: system package is not found or is unusable")
+    endif()
+endif()
+
+if(WITH_GLM STREQUAL "bundled" OR (WITH_GLM STREQUAL "auto" AND NOT glm_FOUND))
+    message(STATUS "GLM version used: BUNDLED")
+
     set(GLM_BASE_PATH "${EXTERNALS_DIR}")
 
     if(NOT EXISTS "${GLM_BASE_PATH}/glm/CMakeLists.txt")
--
libgit2 1.0.1

